mail_notification_custom_subject/models/mail_thread.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2020 Tecnativa - João Marques
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, api
+
+
+class MailThread(models.AbstractModel):
+    _inherit = "mail.thread"
+
+    @api.multi
+    @api.returns('mail.message', lambda value: value.id)
+    def message_post(
+        self, body='', subject=None,
+        message_type='notification', subtype=None,
+        parent_id=False, attachments=None,
+        notif_layout=False, add_sign=True, model_description=False,
+        mail_auto_delete=True, **kwargs
+    ):
+        subtype_id = kwargs.get('subtype_id', False)
+        if not subtype_id:
+            subtype = subtype or 'mt_note'
+            if '.' not in subtype:
+                subtype = 'mail.%s' % subtype
+            subtype_id = self.env['ir.model.data'].xmlid_to_res_id(subtype)
+        mail_templates = self.env["mail.message.custom.subject"].search([
+            ("model_id.model", "=", self._name),
+            ("subtype_ids", "=", subtype_id),
+        ])
+        for template in mail_templates:
+            rendered_subject_template = self.env["mail.template"]._render_template(
+                template_txt=template.subject_template,
+                model=self._name, res_ids=self.id
+            )
```

Comment:
I'm thinking that we shouldn't interrupt by any chance the notification sending:
```suggestion
            try:
                rendered_subject_template = self.env["mail.template"]._render_template(
                    template_txt=template.subject_template,
                    model=self._name, res_ids=self.id
                )
            except Exception:
                rendered_subject_template = False
```

Reply:
Ok, thanks! I just made the change. 
However, can't that make it harder to debug if a subject template is well constructed or not? The `_render_template` function throws some helpful exceptions for that (https://github.com/odoo/odoo/blob/fd56023d26dcebb1503fe8d4ece6d6125dfc3e90/addons/mail/models/mail_template.py#L338). Should I catch them and do something with them?

Reply:
You would already have the log entry, so no, nothing should block the notification.

Reply:
Ok, nice! :+1: